classes/UpgradeContainer.php

```diff
@@ -0,0 +1,466 @@
+                    $upgrader->checkPSVersion(false, array('private', 'minor'));
+                } else {
+                    $upgrader->checkPSVersion(false, array('minor'));
+                }
+        }
+        $this->getState()->setInstallVersion($upgrader->version_num);
+        $this->upgrader = $upgrader;
+        return $this->upgrader;
+    }
+
+    public function getFilesystemAdapter()
+    {
+        if (!is_null($this->filesystemAdapter)) {
+            return $this->filesystemAdapter;
+        }
+
+        $this->filesystemAdapter = new FilesystemAdapter(
+            $this->getFileFilter(),
+            $this->getState()->getRestoreFilesFilename(),
+            $this->getProperty(self::WORKSPACE_PATH),
+            str_replace($this->getProperty(self::PS_ROOT_PATH), '', $this->getProperty(self::PS_ADMIN_PATH)), $this->getProperty(self::PS_ROOT_PATH));
+
+        return $this->filesystemAdapter;
+    }
+
+    /**
+     * @return Logger
+     */
+    public function getLogger()
+    {
+        if (! is_null($this->logger)) {
+            return $this->logger;
+        }
+
+        $logFile = $this->getProperty(self::TMP_PATH).DIRECTORY_SEPARATOR.'log.txt';
+        $this->logger = new LegacyLogger($logFile);
+        return $this->logger;
+    }
+
+    public function setLogger(Logger $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function getModuleAdapter()
+    {
+        if (!is_null($this->moduleAdapter)) {
+            return $this->moduleAdapter;
+        }
+
+        $this->moduleAdapter = new ModuleAdapter(
+            $this->getDb(),
+            $this->getTranslator(),
+            $this->getProperty(self::PS_ROOT_PATH).DIRECTORY_SEPARATOR.'modules'.DIRECTORY_SEPARATOR,
+            $this->getProperty(self::TMP_PATH),
+            $this->getState()->getInstallVersion(),
+            $this->getZipAction());
+
+        return $this->moduleAdapter;
+    }
+
+    public function getState()
+    {
+        if (!is_null($this->state)) {
+            return $this->state;
+        }
+
+        $this->state = new State();
+        return $this->state;
+    }
+
+    public function getTranslationAdapter()
+    {
+        return new Translation($this->getTranslator(), $this->getLogger(), $this->getState()->getInstalledLanguagesIso());
+    }
+
+    public function getTranslator()
+    {
+        return new \PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator('AdminSelfUpgrade');
+    }
+
+    public function getTwig()
+    {
+        if (!is_null($this->twig)) {
+            return $this->twig;
+        }
+
+        // Using independant template engine for 1.6 & 1.7 compatibility
+        $loader = new \Twig_Loader_Filesystem();
+        $loader->addPath(realpath(__DIR__.'/..').'/views/templates', 'ModuleAutoUpgrade');
+        $twig = new \Twig_Environment($loader, array(
+            //'cache' => '/path/to/compilation_cache',
```

Comment:
use namespace and remove useless comments

Reply:
Even better, I will enable the cache from now. I couldn't before because getting access to some path was difficult.